modules/light-clients/06-solomachine/types/client_state_test.go

```diff
@@ -845,3 +846,57 @@ func (suite *SoloMachineTestSuite) TestVerifyNextSeqRecv() {
 		}
 	}
 }
+
+func (suite *SoloMachineTestSuite) TestGetTimestampAtHeight() {
+	tmPath := ibctesting.NewPath(suite.chainA, suite.chainB)
+	suite.coordinator.SetupClients(tmPath)
+
+	testCases := []struct {
+		name        string
+		clientState *types.ClientState
+		// Write directly to store since solomachine isn't hooked up to the test app.
+		malleate func()
+		expValue uint64
+		expPass  bool
+	}{
+		{
+			name:        "get timestamp at height exists",
+			clientState: suite.solomachine.ClientState(),
+			malleate: func() {
+				value, err := clienttypes.MarshalConsensusState(suite.chainA.Codec, suite.solomachine.ClientState().ConsensusState)
+				suite.Require().NoError(err)
+				suite.store.Set(host.ConsensusStateKey(consensusHeight), value)
+			},
+			expValue: suite.solomachine.ClientState().ConsensusState.Timestamp,
+			expPass:  true,
+		},
+		{
+			name:        "get timestamp at height not exists",
+			clientState: suite.solomachine.ClientState(),
+			malleate:    func() {},
+		},
+	}
```

Comment:
test should ensure the timestamp returned matches the consensus state timestamp (stored in the client state)